modules/core/shared/src/main/scala/coursier/core/Dependency.scala

```diff
@@ -45,6 +45,16 @@ import dataclass.data
   def withPublication(name: String, `type`: Type, ext: Extension, classifier: Classifier): Dependency =
     withPublication(Publication(name, `type`, ext, classifier))
 
+  def copy(
```

Comment:
Like above, let's just add a more fine grained `update` method, with no default values.

Reply:
Just like above, we can mark this `private[core]`, so that we don't have to maintain binary compatibility for it:
```suggestion
  private[core] def copy(
```